contrib/packs/tests/test_action_download.py

```diff
@@ -289,3 +289,58 @@ def test_resolve_urls(self):
 
         url = DownloadGitRepoAction._eval_repo_url("file:///home/vagrant/stackstorm-test")
         self.assertEqual(url, "file:///home/vagrant/stackstorm-test")
+
+    def test_run_pack_download_edge_cases(self):
```

Comment:
Thanks for working on the tests.

And yeah, I knew that sadly testing will be quite a hassle because it's hard to replicate a default branch problem without mocking things out...

I will also look into if it's possible to write some more tests which require less mocking so we can exercise more code paths...